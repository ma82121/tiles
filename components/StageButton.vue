<template>
  <div
    :class="['stage-button', isDisabled ? 'is-disabled' : '']"
    @click="click"
  >
    <div v-show="stageNumber > -1" class="stage-button-head">
      {{ stageNumber + 1 }}
    </div>
    <div class="stage-button-body">
      <div class="stage-button-board">
        <div
          v-for="tile in stage"
          :key="tile.id"
          :class="['tile', 'is-' + tile.type, tile.isOpen ? 'is-open' : '']"
        />
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'StageButton',
  props: {
    stage: {
      type: Array,
      default: () => []
    },
    stageNumber: {
      type: Number,
      default: -1
    },
    isDisabled: {
      type: Boolean,
      default: false
    }
  },
  methods: {
    click(e) {
      this.$emit('click', e)
    }
  }
}
</script>

<style lang="scss" scoped>
.stage-button {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;

  border-radius: 4px;
  padding: 5px;
  cursor: pointer;
  color: #fff;
  background-color: #1cc4fd;

  width: 80px;
  height: 80px;

  &.is-disabled {
    background-color: #444;
    cursor: default;

    .tile {
      display: none;
    }
  }

  .stage-button-head {
    margin: -3px 0 3px;
    text-align: center;
    font-size: 14px;
    font-weight: bold;
    line-height: 1;
  }

  .stage-button-body {
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .stage-button-board {
    position: relative;
    width: 50px;
    height: 50px;
  }

  .tile {
    position: absolute;
    width: 11px;
    height: 11px;
    border-radius: 2px;
    box-sizing: border-box;
    border: 1px solid #999;

    @for $i from 0 through 24 {
      &:nth-of-type(#{$i + 1}) {
        top: 10px * floor($i / 5);
        left: 10px * ($i % 5);
      }
    }

    &.is-1 {
      background-color: #222;
    }

    &.is-2 {
      background-color: #0f0;
    }

    &.is-3 {
      background-color: #f00;
    }

    &.is-4 {
      background-color: #ff0;
    }

    &.is-open {
      &.is-1,
      &.is-2,
      &.is-3,
      &.is-4 {
        background-color: #fff;
      }
    }
  }
}
</style>
