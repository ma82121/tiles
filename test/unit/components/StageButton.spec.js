import { shallowMount } from '@vue/test-utils'
import StageButton from '@/components/StageButton'

describe('StageButton', () => {
  const wrapper = shallowMount(StageButton, {
    propsData: {
      stage: [
        { index: 0, type: '1', isOpen: false },
        { index: 1, type: '2', isOpen: false },
        { index: 2, type: '3', isOpen: false },
        { index: 3, type: '4', isOpen: false },
        { index: 4, type: '1', isOpen: false },
        { index: 5, type: '1', isOpen: false },
        { index: 6, type: '1', isOpen: false },
        { index: 7, type: '1', isOpen: true },
        { index: 8, type: '1', isOpen: false },
        { index: 9, type: '1', isOpen: false },
        { index: 10, type: '1', isOpen: false },
        { index: 11, type: '1', isOpen: false },
        { index: 12, type: '1', isOpen: false },
        { index: 13, type: '1', isOpen: false },
        { index: 14, type: '1', isOpen: false },
        { index: 15, type: '1', isOpen: false },
        { index: 16, type: '1', isOpen: false },
        { index: 17, type: '1', isOpen: true },
        { index: 18, type: '1', isOpen: false },
        { index: 19, type: '1', isOpen: false },
        { index: 20, type: '1', isOpen: false },
        { index: 21, type: '1', isOpen: false },
        { index: 22, type: '1', isOpen: false },
        { index: 23, type: '1', isOpen: false },
        { index: 24, type: '1', isOpen: false }
      ]
    }
  })

  test('snapshot', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
