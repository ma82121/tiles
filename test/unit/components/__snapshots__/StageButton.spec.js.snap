// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageButton snapshot 1`] = `
<div
  class="stage-button"
>
  <div
    class="stage-button-head"
    style="display: none;"
  >
    
    0
  
  </div>
   
  <div
    class="stage-button-body"
  >
    <div
      class="stage-button-board"
    >
      <div
        class="tile is-1"
      />
      <div
        class="tile is-2"
      />
      <div
        class="tile is-3"
      />
      <div
        class="tile is-4"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1 is-open"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1 is-open"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
      <div
        class="tile is-1"
      />
    </div>
  </div>
</div>
`;
